src/inspector_agent.cc

```diff
@@ -259,8 +286,26 @@ class ChannelImpl final : public v8_inspector::V8Inspector::Channel,
             Utf8ToStringView(method)->string())) {
       session_->dispatchProtocolMessage(message);
     } else {
-      node_dispatcher_->dispatch(call_id, method, std::move(value),
-                                 raw_message);
+      node_dispatcher_->dispatch(
+          call_id, method, std::move(value), raw_message);
+    }
+    // Since the `Network` domain is not directly accessible to inspector
+    // module users, this allows for them to enable/disable the `Network`
+    // domain via the `NodeNetwork` domain.
+    if (method == "NodeNetwork.enable") {
+      network_agent_->enable();
+    }
+    if (method == "NodeNetwork.disable") {
+      network_agent_->disable();
+    }
+    // Since DevTools Frontend sends commands in the `Network` domain only,
+    // we need to enable/disable the `NodeNetwork` domain via the `Network`
+    // domain.
+    if (method == "Network.enable") {
+      node_network_agent_->enable();
+    }
+    if (method == "Network.disable") {
+      node_network_agent_->disable();
```

Comment:
This approach looks a bit messy... I'm thinking I should create a class that manages both `NetworkAgent` and `NodeNetworkAgent` and handles their interactions (such as syncing enable/disable states and notifying each other of events).

Reply:
Refactored in https://github.com/nodejs/node/pull/53593/commits/4103e26eb9d79a25ae7a53c0b67a6d5701910545